### PR TITLE
Image data

### PR DESCRIPTION
**Pull request by ibgreen**
- Replace `options.image.type: 'ndarray'` for Node.js with `options.image.type: 'data'`
- Support `options.image.type: 'data'` this on both Node.js and browser. 
- Continued cleanup and consolidation of the extensive image category docs

Apart from unifying the API and making it easy to load image data for image textures, also enables moving pixel extraction into the image worker (next PR), increasing parallel throughput of data texture loading.

### CODE REVIEW

File: modules/images/src/lib/parsed-image-api/image-type.js [hunk 1]
```diff
@@ -1,42 +1,48 @@
...
+const IMAGE_SUPPORTED = typeof Image !== 'undefined'; // NOTE: "false" positives if jsdom is installed
+const IMAGE_BITMAP_SUPPORTED = typeof ImageBitmap !== 'undefined';
+const NODE_IMAGE_SUPPORTED = Boolean(global._parseImageNode);
 
 // Checks if a loaders.gl image type is supported
 export function isImageTypeSupported(type) {
   switch (type) {
     case 'auto':
-      return true;
+      // Should only ever be false in Node.js, if polyfills have not been installed...
```
Comment (xintongxia) [inline modules/images/src/lib/parsed-image-api/image-type.js]: maybe notify the users that they could potentially install `@loaders.gl/polyfill` package?

**Comment by Pessimistress** [comment]
I'd prefer `type: 'array'` or `type: 'typedarray`. `data` is so generic the user will not know what format to expect.

**Comment by ibgreen** [comment]
> I'd prefer type: 'array' or type: 'typedarray. data is so generic the user will not know what format to expect.

Yes... some of my thoughts:

- The intended emphasis was between loading as an opaque image (`Image`/`ImageBitmap`) and addressable data, not emphasizing the exact type of the returned data.
- Given that it is an object with `ImageData` fields containing a typed array, not actually a typed array, does calling it `typedarray` still make sense? I initially called it `imagedata` but changed it for fear of confusing with the actual HTML5 type, but I still like that variant.
- Another wrinkle is that the array can represent a compressed image, so sometimes not easily application accessible....

I am still not decided on whether to merge this. I almost have the benchmark infrastructure in place to validate that this is worthwhile. If it pans out I will explain the thinking more fully and get your thoughts on the approach.

**Comment by ibgreen** [comment]
@Pessimistress Due to complications with the custom SVG loading flow, the `options.image.type: data` currently does not work for SVG files, so I removed those test cases. This seems acceptable for now, as I would expect it to be rare to store data textures in SVG format.

**Pull request merged**